controllers/nginx/pkg/cmd/controller/nginx.go

```diff
@@ -256,7 +257,12 @@ func (n NGINXController) Info() *ingress.BackendInfo {
 
 // OverrideFlags customize NGINX controller flags
 func (n NGINXController) OverrideFlags(flags *pflag.FlagSet) {
-	flags.Set("ingress-class", "nginx")
+	flags.Set("ingress-class", defIngressClass)
```

Comment:
@gianrubio add a warning if `flag != ""` or `flag != "nginx"` saying that only ingress rules with the class `flag value` will be processed